### PR TITLE
feat(alerting): Add group-specific WebHook URL for Slack

### PR DESCRIPTION
**Pull request by mani9223-oss**

### CODE REVIEW

**Comment by mani9223-oss** [comment]
@TwiN Please check. Fixed indent issues

File: README.md [hunk 1]
```diff
@@ -626,11 +626,19 @@ endpoints:
 | `alerting.slack`               | Configuration for alerts of type `slack`                                                   | `{}`          |
 | `alerting.slack.webhook-url`   | Slack Webhook URL                                                                          | Required `""` |
 | `alerting.slack.default-alert` | Default alert configuration. <br />See [Setting a default alert](#setting-a-default-alert) | N/A           |
-
+| `alerting.slack.overrides`               | List of overrides that may be prioritized over the default configuration                   | `[]`          |
+| `alerting.slack.overrides[].group`       | Endpoint group for which the configuration will be overridden by this configuration        | `""`          |
+| `alerting.slack.overrides[].webhook-url` | Slack Webhook URL                                                                        | `""`          |
```
Comment (TwiN) [inline README.md:631]: Could you please fix the table format so that all columns are of the same size? 🙏

File: alerting/provider/slack/slack.go [hunk 2]
```diff
@@ -5,7 +5,6 @@ import (
 	"fmt"
 	"io"
 	"net/http"
-
 	"github.com/TwiN/gatus/v3/alerting/alert"
```
Comment (TwiN) [inline alerting/provider/slack/slack.go:8]: Could you undo this change?

File: alerting/provider/slack/slack_test.go [hunk 3]
```diff
@@ -21,7 +20,42 @@ func TestAlertProvider_IsValid(t *testing.T) {
 		t.Error("provider should've been valid")
 	}
 }
-
+func TestAlertProvider_IsValidWithOverride(t *testing.T) {
```
Comment (TwiN) [inline alerting/provider/slack/slack_test.go:23]: Could you re-add a newline between these two?

**Review by TwiN — CHANGES_REQUESTED** [review]

**Comment by mani9223-oss** [comment]
@TwiN Changes are done

**Review by TwiN — CHANGES_REQUESTED** [review]
It appears like the Go code you've added is double indented; could you fix it please? 🙏

File: alerting/provider/slack/slack.go [hunk 4]
```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 	"io"
 	"net/http"
-
```
Comment (appleboy) [inline alerting/provider/slack/slack.go:8]: redundant tab

**Review by TwiN — APPROVED** [review]

File: alerting/provider/slack/slack_test.go [hunk 5]
```diff
@@ -105,7 +141,6 @@ func TestAlertProvider_buildRequestBody(t *testing.T) {
 	scenarios := []struct {
 		Name         string
 		Provider     AlertProvider
-		Endpoint     core.Endpoint
```
Comment (TwiN) [inline alerting/provider/slack/slack_test.go:108]: Why did you remove this? 🤔 Some tests seem to be failing because of this

**Review by TwiN — CHANGES_REQUESTED** [review]

**Comment by Coolomina** [comment]
👋🏻  Why are we choosing this specific implementation when we can have channel override instead of webhook_url override?

That way we could have several groups benefitting from the same webhook_url. Easier to maintain, easier to rotate. What do you think ?

**Comment by TwiN** [comment]
> 👋🏻 Why are we choosing this specific implementation when we can have channel override instead of webhook_url override?
> 
> That way we could have several groups benefitting from the same webhook_url. Easier to maintain, easier to rotate. What do you think ?, I guess it would be a matter of modifying the request body https://github.com/TwiN/gatus/blob/master/alerting/provider/slack/slack.go#L70
> 
> (edit): There are organisations that rely on Security team to configure these kinds of configuration, as this can easily be treated as a secret, so if you were to have 200 apps, having 200 slack webhooks sounds terrible.

Hello @Coolomina, last I checked, channel overrides do not work on Slack; this is why we don't use channel overrides. 

Well, Slack's legacy "Incoming webhooks" app does still support it, but I'm not sure I want to build features around "deprecated" features, as that would lead to confusion.

[source](https://api.slack.com/legacy/custom-integrations/messaging/webhooks)

If you look at the non-legacy counterpart [here](https://api.slack.com/messaging/webhooks), they make no mention of channel overrides.

**Comment by Coolomina** [comment]
Thanks for the answer @TwiN :)

I totally agree with you in not building features for legacy integrations, I just did not know this was on its way to deprecation. 

I still think it does not scale as it's very easy for an org to have +5 delivery teams with their own alerting channels. I'll contact Slack support and see what the best practice is here.

Thank you again!

**Comment by Coolomina** [comment]
@TwiN @mani9223-oss need any help with the PR?. Happy to help.

**Review by TwiN — APPROVED** [review]

**Comment by TwiN** [comment]
Thank you for the contribution @mani9223-oss!

**Comment by TwiN** [comment]
Released in [v4.0.0](https://github.com/TwiN/gatus/releases/tag/v4.0.0)

**Pull request merged**